src/fsharp/ConstraintSolver.fs

```diff
@@ -1132,7 +1137,15 @@ and SolveTypeSubsumesType (csenv: ConstraintSolverEnv) ndeep m2 (trace: Optional
 
     | TType_ucase (uc1, l1), TType_ucase (uc2, l2) when g.unionCaseRefEq uc1 uc2  -> 
         SolveTypeEqualsTypeEqns csenv ndeep m2 trace cxsln l1 l2
-
+        
+    | TType_erased_union (_,l1), TType_erased_union (_,l2) when typeAEquiv g aenv sty1 sty2 ->
```

Comment:
This rule needs much work - it's currently implementing equality not subsumption, and indeed exact equality without solution because of `typeAEquiv g aenv sty1 sty2`